var/spack/repos/builtin/packages/ccfits/package.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Ccfits(Package):
+    """CCfits is an object oriented interface to the cfitsio library.
+    It is designed to make the capabilities of cfitsio available to programmers
+    working in C++. It is written in ANSI C++ and implemented using the C++
+    Standard Library with namespaces, exception handling, and member template
+    functions. """
+
+    homepage = "https://heasarc.gsfc.nasa.gov/fitsio/CCfits/"
+    url      = "https://heasarc.gsfc.nasa.gov/fitsio/CCfits/CCfits-2.5.tar.gz"
+
+    version('2.5', sha256='938ecd25239e65f519b8d2b50702416edc723de5f0a5387cceea8c4004a44740')
+    version('2.4', sha256='ba6c5012b260adf7633f92581279ea582e331343d8c973981aa7de07242bd7f8')
+
+    depends_on('cfitsio')
+
+    def url_for_version(self, version):
+        url = 'https://heasarc.gsfc.nasa.gov/fitsio/CCfits/CCfits-{0}.tar.gz'
+        return url.format(version)
+
+    def install(self, spec, prefix):
+        print("Install ... ")
```

Comment:
@grisu48 

* This appears to fit the template of a `MakefilePackage`, if you inherit from that class (rather than `Package`) you should not need to define an `install` method at all.
* Note that spack will generally print a message like "Installing <package name>" already.
* The URL format used by this package is simple enough that it appears that Spack's default URL version substitution should be sufficient

None of these are especially problematic. If you'd like to make edits then I can take another look after. Otherwise let me know and this can be merged now.

Reply:
Thanks for the reply, I will work on the suggestions tomorrow and incorporate them. Just came back from a short holiday.